### PR TITLE
server: support decode column

### PR DESCRIPTION
**Pull request by zimulala**
The corresponding column value is decoded by column ID, column type, column flag, column length and row value.

I will add tests to this PR later. So I added the label of `DNM`

### CODE REVIEW

File: server/region_handler.go [hunk 1]
```diff
@@ -137,6 +148,67 @@ func (s *Server) newRegionHandler() (hanler *regionHandler) {
...
+	// Construct field type.
+	ft := &types.FieldType{
+		Tp:   byte(colTp),
+		Flag: uint(colFlag),
+		Flen: int(colLen),
+	}
+	// Decode a column.
+	m := make(map[int64]*types.FieldType, 1)
+	m[int64(colID)] = ft
+	loc := time.Now().Location()
```
Comment (jackysp) [inline server/region_handler.go]: Maybe UTC is better.

**Comment by zimulala** [comment]
PTAL @jackysp

File: server/region_handler_test.go [hunk 2]
```diff
@@ -327,6 +332,56 @@ func (ts *TidbRegionHandlerTestSuite) TestGetMvccNotFound(c *C) {
...
+	colIDs := make([]int64, 0, 3)
+	for _, col := range cols {
+		colIDs = append(colIDs, col.id)
+	}
+	sc := &stmtctx.StatementContext{TimeZone: time.UTC}
+	bs, err := tablecodec.EncodeRow(sc, row, colIDs, nil, nil)
+	c.Assert(err, IsNil)
+	c.Assert(bs, NotNil)
+
+	unitTest := func(col *column) {
```
Comment (jackysp) [inline server/region_handler_test.go]: why use a function object here?

**Comment by jackysp** [comment]
It's better to update https://github.com/pingcap/tidb/blob/master/docs/tidb_http_api.md in this pr.

File: server/region_handler_test.go (hunk 2 above)
Comment (zimulala) [reply server/region_handler_test.go]: I think it's OK.

**Comment by zimulala** [comment]
PTAL @jackysp

**Comment by jackysp** [comment]
LGTM
PTAL @AndreMouche

**Comment by zz-jason** [comment]
@zimulala plz resolve conflicts.

**Pull request merged**